requre/cassette.py

```diff
@@ -19,16 +19,716 @@
+    def metadata(self):
+        return self.data.metadata
+
+    @metadata.setter
+    def metadata(self, value):
+        self.data.metadata = value
+
+
+class Cassette:
+    """
+    Class implements reading/writing simple JSON requests to dict structure
+    and return values based on keys.
+    It contains methods to reads/stores data to object and load and store them to YAML file
+
+    storage_object: dict with structured data based on keys (eg. simple JSON requests)
+    storage_file: file for reading and writing data in storage_object
+    """
 
     @property
     def content(self) -> dict:
-        return PersistentObjectStorage().storage_object
+        return self.storage_object
+
+    internal_object_key = METATADA_KEY
+    version_key = "version_storage_file"
+    key_inspect_strategy_key = "key_strategy"
+
+    def _set_defaults(self) -> None:
+        self.dump_after_store = False
+        storage_mode = os.getenv(ENV_REQURE_STORAGE_MODE, "default")
+        logger.info(
+            f"You overrided storage mode via env var {ENV_REQURE_STORAGE_MODE}={storage_mode}"
+        )
+        if not hasattr(StorageMode, storage_mode):
+            raise PersistentStorageException(
+                f"storage mode '{storage_mode}' does not exist, "
+                f"use one of {list(StorageMode.__members__.keys())}) "
+            )
+        self.mode: StorageMode = getattr(StorageMode, storage_mode)
+        self.is_flushed = False
+        self.storage_object: dict = {}
+        self._storage_file: Optional[str] = None
+        self.data_miner = DataMiner()
+
+    def __init__(self) -> None:
+        # call dump() after store() is called
+        self._set_defaults()
+        storage_file_from_env = os.getenv(ENV_STORAGE_FILE)
+        if storage_file_from_env:
+            self.storage_file = storage_file_from_env
+
+    def __str__(self) -> str:
+        return (
+            f"{self.__class__.__name__}("
+            f"storage_mode={self.mode}, "
+            f"storage_file={self._storage_file})"
+        )
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    @property
+    def metadata(self):
+        """
+        Placeholder to store some custom configuration, or store custom data,
+        eg. version of storage file, or some versions of packages
+
+        :return: dict
+        """
+        return self.storage_object.get(self.internal_object_key, {})
+
+    @metadata.setter
+    def metadata(self, key_dict: dict):
+        if self.internal_object_key not in self.storage_object:
+            self.storage_object[self.internal_object_key] = {}
+        for k, v in key_dict.items():
+            self.storage_object[self.internal_object_key][k] = v
+        self.is_flushed = False
+
+    @property
+    def storage_file_version(self):
+        """
+        Get version of persistent storage file
+        :return: int
+        """
+        return self.metadata.get(self.version_key, 0)
+
+    def do_store(self, keys):
```

Comment:
What is the difference and relation between `store` and `do_store`?

Reply:
yes, please do as much as possible as good first issues  there.

function with `do_store` was written by you and not sure if there is any benefit. So we can merge it in new PRs.

Reply:
OK, I can't remember..;)